awx/main/access.py

```diff
@@ -1675,7 +1698,7 @@ def can_attach(self, obj, sub_obj, relationship, data, skip_sub_obj_read_check=F
         if relationship == "instance_groups":
             if not obj.organization:
                 return False
-            return self.user.can_access(type(sub_obj), "read", sub_obj) and self.user in obj.organization.admin_role
+            return self.user in sub_obj.use_role
```

Comment:
My mind is going some accounting here - "How many objects can we attach an IG to?" I think the answer is 3.

For each one of those I expect to see the `use_role` check in `can_attach`. Right now I see 2 in this change.

So now working backwards... we attach IGs to JTs, organizations, inventories.

I think you still need to update organization access.

Reply:
I left the organization access alone because a regular user shouldn't be able to change an organization anyway. If you would like I can try adjusting it but I didn't want to make any changes that would fundamentally change the rbac.

Reply:
Okay, I'm not sure, I need to talk that out.

https://github.com/ansible/awx/blob/60356afeec61ad8d81ba5e7cfb6ca1545fbd2d90/awx/main/access.py#L864

https://github.com/ansible/awx/blob/60356afeec61ad8d81ba5e7cfb6ca1545fbd2d90/awx/main/access.py#L870-L873

Here, `obj` is the organization object and `sub_obj` is the instance group. The request is to add an instance group to the organization's instance group list, which is a fallback for jobs (does not delegate IG permission anymore).

I'm not seeing a sufficiently strong justification for the new pattern you're developing here, which would suggest a requirement of "admin" to `obj` (org) and "use" to `sub_obj` (IG).